cli/internal/runsummary/vercel.go

```diff
@@ -22,28 +26,91 @@ type vercelRunPayload struct {
+	// The command that kicked off the turbo run
+	Command string `json:"command,omitempty"`
+
+	// RepositoryPath is the relative directory from the turborepo root to where
+	// the command was invoked.
+	RepositoryPath string `json:"repositoryPath,omitempty"`
+
+	// Context is the host on which this Run was executed (e.g. Vercel)
+	Context string `json:"context,omitempty"`
+
 	// TODO: we need to add these in
 	// originationUser string
 	// gitBranch       string
 	// gitSha          string
-	// context         string
-	// command         string
 }
 
-func newVercelRunCreatePayload(runsummary *RunSummary) *vercelRunPayload {
-	startTime := int(runsummary.ExecutionSummary.startedAt.UnixMilli())
+type vercelCacheStatus struct {
+	Status string `json:"status,omitempty"`
+	Source string `json:"source,omitempty"`
+}
+
+type vercelTask struct {
+	Key          string            `json:"key,omitempty"`
+	Name         string            `json:"name,omitempty"`
+	Workspace    string            `json:"workspace,omitempty"`
+	Hash         string            `json:"hash,omitempty"`
+	StartTime    int64             `json:"startTime,omitempty"`
+	EndTime      int64             `json:"endTime,omitempty"`
+	Cache        vercelCacheStatus `json:"cache,omitempty"`
+	ExitCode     int               `json:"exitCode,omitempty"`
+	Dependencies []string          `json:"dependencies,omitempty"`
+	Dependents   []string          `json:"dependents,omitempty"`
+}
+
+func (rsm *Meta) newVercelRunCreatePayload() *vercelRunPayload {
+	startTime := rsm.RunSummary.ExecutionSummary.startedAt.UnixMilli()
+	context := "LOCAL"
+	if name := ci.Constant(); name != "" {
+		context = name
+	}
 	return &vercelRunPayload{
-		StartTime: startTime,
-		Status:    "started",
-		Type:      "TURBO",
+		StartTime:      startTime,
+		Status:         "running",
+		Command:        rsm.synthesizedCommand,
+		RepositoryPath: rsm.repoPath.ToString(),
+		Type:           "TURBO",
+		Context:        context,
 	}
 }
 
 func newVercelDonePayload(runsummary *RunSummary) *vercelRunPayload {
-	endTime := int(runsummary.ExecutionSummary.endedAt.UnixMilli())
+	endTime := runsummary.ExecutionSummary.endedAt.UnixMilli()
 	return &vercelRunPayload{
 		Status:   "completed",
 		EndTime:  endTime,
 		ExitCode: runsummary.ExecutionSummary.exitCode,
 	}
 }
+
+func newVercelTaskPayload(taskSummary *TaskSummary) *vercelTask {
+	hit := taskSummary.CacheState.Local || taskSummary.CacheState.Remote
+	status := "MISS"
+	var source string
+	if hit {
+		source = "REMOTE"
+		status = "HIT"
+	}
+
+	startTime := taskSummary.Execution.startAt.UnixMilli()
+	endTime := taskSummary.Execution.endTime().UnixMilli()
+
+	return &vercelTask{
+		Key:       taskSummary.TaskID,
+		Name:      taskSummary.Task,
+		Workspace: taskSummary.Package,
+		Hash:      taskSummary.Hash,
+		StartTime: startTime,
+		EndTime:   endTime,
+		Cache: vercelCacheStatus{
+			Status: status,
+			Source: source,
+		},
+		ExitCode:     *taskSummary.Execution.exitCode,
```

Comment:
I'm finding some exitCodes for tasks are not being set.

Reply:
they won't be set for failed tasks. that will be covered in #4524.